integration-tests/src/test/java/com/datastax/oss/driver/core/resolver/MockResolverIT.java

```diff
@@ -58,52 +61,290 @@
+          if (nodes.size() == 3) {
+            break;
+          }
+          Thread.sleep(1000);
+        } catch (InterruptedException e) {
+          break;
+        }
+      }
       ResultSet rs = session.execute("SELECT * FROM system.local");
-      List<Row> rows = rs.all();
-      assertThat(rows).hasSize(1);
-      LOG.trace("system.local contents: {}", rows.get(0).getFormattedContents());
+      assertThat(rs).isNotNull();
+      Row row = rs.one();
+      assertThat(row).isNotNull();
       Collection<Node> nodes = session.getMetadata().getNodes().values();
-      for (Node node : nodes) {
-        LOG.trace("Found metadata node: {}", node);
+      assertThat(nodes).hasSize(3);
+      Iterator<Node> iterator = nodes.iterator();
+      while (iterator.hasNext()) {
+        LOG.trace("Metadata node: " + iterator.next().toString());
       }
       Set<Node> filteredNodes;
       filteredNodes =
           nodes.stream()
-              .filter(x -> x.toString().contains("node-1.cluster.fake"))
+              .filter(x -> x.toString().contains("test.cluster.fake"))
               .collect(Collectors.toSet());
       assertThat(filteredNodes).hasSize(1);
-      InetSocketAddress address =
-          (InetSocketAddress) filteredNodes.iterator().next().getEndPoint().resolve();
-      assertTrue(address.isUnresolved());
     }
+    try (CcmBridge ccmBridge = CcmBridge.builder().withNodes(3).withIpPrefix("127.0.1.").build()) {
+      ccmBridge.create();
+      ccmBridge.start();
+      for (int i = 0; i < 30; i++) {
+        try {
+          Collection<Node> nodes = session.getMetadata().getNodes().values();
+          if (nodes.size() == 3) {
+            break;
+          }
+          Thread.sleep(1000);
+        } catch (InterruptedException e) {
+          break;
+        }
+      }
+      ResultSet rs = session.execute("SELECT * FROM system.local");
+      assertThat(rs).isNotNull();
+      Row row = rs.one();
+      assertThat(row).isNotNull();
+
+      Collection<Node> nodes = session.getMetadata().getNodes().values();
+      assertThat(nodes).hasSize(3);
+      Iterator<Node> iterator = nodes.iterator();
+      while (iterator.hasNext()) {
+        LOG.trace("Metadata node: " + iterator.next().toString());
+      }
+      Set<Node> filteredNodes;
+      filteredNodes =
+          nodes.stream()
+              .filter(x -> x.toString().contains("test.cluster.fake"))
+              .collect(Collectors.toSet());
+      if (filteredNodes.size() == 0) {
+        LOG.error(
+            "No metadata node with \"test.cluster.fake\" substring. The unresolved endpoint socket was likely "
+                + "replaced with resolved one.");
+      } else if (filteredNodes.size() > 1) {
+        fail(
+            "Somehow there is more than 1 node in metadata with unresolved hostname. This should not ever happen.");
+      }
+    }
+    session.close();
+  }
+
+  @Test
+  public void run_replace_test_20_times() {
```

Comment:
You may need this test locally if `replace_cluster_test` but we definitely don't want to have this test in the repo

Reply:
Moved annotation from this to `replace_cluster_test` so that it runs once and only logs an error if something happens.

Reply:
I need to fix the conflicts before rebasing this branch though

Reply:
fixed